test/managed_scip.jl

```diff
@@ -0,0 +1,45 @@
+# Test memory management
+
+@testset "create and manual free" begin
+    mscip = SCIP.ManagedSCIP()
+    @test mscip.scip[] != C_NULL
+    SCIP.free_scip(mscip)
+    @test mscip.scip[] == C_NULL
+end
+
+@testset "create and semi-manual free" begin
+    mscip = SCIP.ManagedSCIP()
+    @test mscip.scip[] != C_NULL
+    finalize(mscip)
+    @test mscip.scip[] == C_NULL
+end
+
+@testset "create and implicit free" begin
+    ptr = Ref{Ptr{SCIP.SCIP_}}(C_NULL)
+    for i = 1:1
+        mscip = SCIP.ManagedSCIP()
+        ptr = mscip.scip
+        @test ptr[] != C_NULL
+    end
+    GC.gc()
+    GC.gc() # TODO: why need a second call?
```

Comment:
I think there's no guarantee when `finalize` runs. It's better just test that explicit `finalize` works. Depending on the behavior of `GC.gc()` could break the test in different julia versions.